### PR TITLE
Added get patient by ID feature

### PR DESCRIPTION
**Pull request by SambitKSahu99**
Added getPatientByID method in existing service and controller class and made some changes in Patient DTO and Model class

### CODE REVIEW

File: src/main/java/com/theelixrlabs/healthcare/constants/PatientConstants.java [hunk 1]
```diff
@@ -1,13 +1,21 @@
 package com.theelixrlabs.healthcare.constants;
 
+/**
+ * Contains the string constants used for the patient module implementation
+ */
 public class PatientConstants {
     public static final String AADHAAR_FORMAT_PATTERN = "%s %s %s";
+    public static final String AADHAAR_NUMBER_REGEX = "^[2-9][0-9]{11}";
     public static final String AADHAAR_NUMBER_ALREADY_EXISTS = "Aadhaar Number already exists.";
```
Comment (EXR-38) [inline src/main/java/com/theelixrlabs/healthcare/constants/PatientConstants.java]: Please move the validation messages to messages.properties file

File: src/main/java/com/theelixrlabs/healthcare/controller/PatientController.java [hunk 2]
```diff
@@ -8,9 +8,9 @@
 import jakarta.validation.Valid;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
Comment (EXR-38) [inline src/main/java/com/theelixrlabs/healthcare/controller/PatientController.java]: Remove the wild card imports

File: src/main/java/com/theelixrlabs/healthcare/controller/PatientController.java [hunk 3]
```diff
@@ -30,12 +30,30 @@ public PatientController(PatientService patientService) {
...
+     * @param id Patient ID as String
+     * @return ResponseEntity containing a SuccessResponse with the Patient Associated with the ID and HTTP status 200 (OK)
+     * @throws CustomException if validation fails or if there are errors during getting patient through ID.
+     */
+    @GetMapping(PatientConstants.GET_PATIENT_BY_ID_ENDPOINT)
+    public ResponseEntity<?> getPatientById(@PathVariable String id) throws CustomException {
+        UUID patientId;
+        try {
+            patientId = UUID.fromString(id);
+        } catch (Exception exception) {
```
Comment (EXR-38) [inline src/main/java/com/theelixrlabs/healthcare/controller/PatientController.java:63]: Better to specify the exception you are trying to catch here. You can add another catch block to account for the generic exceptions.

File: src/main/java/com/theelixrlabs/healthcare/service/PatientService.java [hunk 4]
```diff
@@ -25,44 +27,44 @@ public PatientService(PatientRepository patientRepository) {
...
-     * @param patientDTO    The data transfer object containing patient information.
-     * @throws CustomException    If validation fails (e.g., empty first name or invalid characters).
+     * @param patientDTO The data transfer object containing patient information.
+     * @throws CustomException If validation fails (e.g., empty first name or invalid characters).
      */
     private void validatePatientDTO(PatientDTO patientDTO) throws CustomException {
 
         //Validate first name
-        if (patientDTO.getFirstName().isEmpty()) {
+        if (patientDTO.getPatientFirstName().isEmpty()) {
```
Comment (EXR-38) [inline src/main/java/com/theelixrlabs/healthcare/service/PatientService.java:40]: What if the patientDTO object is null?

File: src/main/java/com/theelixrlabs/healthcare/service/PatientService.java [hunk 5]
```diff
@@ -25,44 +27,44 @@ public PatientService(PatientRepository patientRepository) {
...
         validatePatientDTO(patientDTO);
 
         //Format Aadhaar number
-        String aadhaarNumber = patientDTO.getAadhaarNumber();
+        String aadhaarNumber = patientDTO.getPatientAadhaarNumber();
         String formattedAadhaarNumber = String.format(PatientConstants.AADHAAR_FORMAT_PATTERN, aadhaarNumber.substring(0, 4), aadhaarNumber.substring(4, 8), aadhaarNumber.substring(8, 12));
 
         //Check if aadhaar number already exists.
-        if (patientRepository.findByAadhaarNumber(formattedAadhaarNumber).isPresent()) {
+        if (patientRepository.findByPatientAadhaarNumber(formattedAadhaarNumber).isPresent()) {
```
Comment (EXR-38) [inline src/main/java/com/theelixrlabs/healthcare/service/PatientService.java:71]: Make the return value of findByPatientAadhaarNumber() optional to accomodate the possibility of it returning null.

**Review by EXR-38 — CHANGES_REQUESTED** [review]
Added comments. Please review and make changes to other areas of the code as well according to them.

File: src/main/java/com/theelixrlabs/healthcare/service/PatientService.java (hunk 4 above)
Comment (SambitKSahu99) [reply src/main/java/com/theelixrlabs/healthcare/service/PatientService.java:40]: Before passing the Patient DTO object to service layer , in controller we are validating it and then we are sending.

File: src/main/java/com/theelixrlabs/healthcare/service/PatientService.java (hunk 5 above)
Comment (SambitKSahu99) [reply src/main/java/com/theelixrlabs/healthcare/service/PatientService.java:71]: In repository layer return type of findByPatientAadhaarNumber() is optional

File: src/main/java/com/theelixrlabs/healthcare/constants/PatientConstants.java (hunk 1 above)
Comment (SambitKSahu99) [reply src/main/java/com/theelixrlabs/healthcare/constants/PatientConstants.java]: Resolved

File: src/main/java/com/theelixrlabs/healthcare/controller/PatientController.java (hunk 2 above)
Comment (SambitKSahu99) [reply src/main/java/com/theelixrlabs/healthcare/controller/PatientController.java]: Resolved

File: src/main/java/com/theelixrlabs/healthcare/controller/PatientController.java (hunk 3 above)
Comment (SambitKSahu99) [reply src/main/java/com/theelixrlabs/healthcare/controller/PatientController.java:63]: Resolved

**Review by EXR-38 — APPROVED** [review]

**Review by ashutosh-2609 — APPROVED** [review]

**Pull request merged**